### DISCUSSION TITLE
How to Create Transparent extra-viewport?

### DISCUSSION THREAD

**Discussion by JamboChen** [emilk/egui#4735] [Q&A] [answered]
I'm developing a game utility application using egui, where the main viewport is transparent and always on top. This has been working perfectly when the viewport is the main one, started with `eframe::run_native` and impl the `eframe::App` trait with `clear_color` and `update` functions.

However, I've encountered an issue when trying to create a transparent extra-viewport using `show_viewport_deferred`. While the extra-viewport retains its style, it is no longer transparent. I suspect this is because the `clear_color` function is no longer called for the extra-viewport. The temporary main viewport, which also impl `clear_color` function, becomes transparent instead.

Is there any way to make the transparent extra-viewport drawn with `show_viewport_deferred` ?

**Comment by j-n-f** [comment]
Having the same issue. I think this is a bug.

**Comment by j-n-f** [comment, marked as answer]
@JamboChen looks like it's tracked here: https://github.com/emilk/egui/issues/3632
